tmt/steps/provision/mock.py

```diff
@@ -0,0 +1,505 @@
+
+    def reboot(
+        self,
+        hard: bool = False,
+        command: Optional[Union[Command, ShellScript]] = None,
+        waiting: Optional[Waiting] = None,
+    ) -> bool:
+        # No localhost reboot allowed!
+        self.debug(f"Doing nothing to reboot guest '{self.primary_address}'.")
+        return False
+
+    def push(
+        self,
+        source: Optional[Path] = None,
+        destination: Optional[Path] = None,
+        options: Optional[list[str]] = None,
+        superuser: bool = False,
+    ) -> None:
+        """
+        Thanks to mock's bind-mounting, no file copying is needed
+        """
+
+    def pull(
+        self,
+        source: Optional[Path] = None,
+        destination: Optional[Path] = None,
+        options: Optional[list[str]] = None,
+        extend_options: Optional[list[str]] = None,
+    ) -> None:
+        """
+        Thanks to mock's bind-mounting, no file copying is needed
+        """
+
+    def load(self, data: tmt.steps.provision.GuestData) -> None:
+        super().load(data)
+        self.mock_shell = MockShell(parent=self, config=self.config, rootdir=self.rootdir)
+        self.mock_config = data.mock_config
+
+    def save(self) -> tmt.steps.provision.GuestData:
+        # inherit
+        return super().save()
+
+    def wake(self) -> None:
+        # noop
+        return super().wake()
+
+    def setup(self) -> None:
+        super().setup()
+        # If we are not using bootstrap, then we need to install the package manager.
+        # Otherwise we use overlayfs and the package manager from bootstrap chroot.
+        if self.mock_config['use_bootstrap'] is not True:
+            (
+                self.mock_shell.command_prefix + ['--install', self.mock_config['package_manager']]
+            ).run(cwd=None, logger=self.parent._logger)
+
+
+@tmt.steps.provides_method('mock')
+class ProvisionMock(tmt.steps.provision.ProvisionPlugin[ProvisionMockData]):
+    """
+    Provisioning using mock chroot.
+    """
+
+    _data_class = ProvisionMockData
+    _guest_class = GuestMock
+
+    _thread_safe = True
+
+    # Guest instance
+    _guest = None
+
+    def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
+        """
+        Provision the container
+        """
+
+        super().go(logger=logger)
+
+        # Create a GuestMock instance
+        data = MockGuestData.from_plugin(self)
+
+        # NOTE this may be unnecessary, tmt package manager detection should work fine
+        # but mock already knows what the package manager is...
+        mock_config = mockbuild.config.simple_load_config(data.config)
+        package_manager = "mock-" + mock_config['package_manager']
+
+        # If this provisioning is selected, then we force the use of `mock-` package manager.
+        # NOTE use a global variable instead?
+        tmt.package_managers.find_package_manager(
+            package_manager
```

Comment:
```suggestion
            "mock-" + mock_config['package_manager']
```

Reply:
I guess it fits into the "align the logging with the rest of tmt" bucket I'd like to sort out as a follow up. tmt showing guest facts should print out package manager as well as other bits, I wouldn't want to see individual plugins invent their logging of shared properties and processes.

Reply:
Followup issue created: https://github.com/teemtee/tmt/issues/4097